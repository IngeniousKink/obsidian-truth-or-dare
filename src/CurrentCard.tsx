import * as React from 'react';
import { ParsedCard, Annotation } from './parse-card.js';
import { useApp } from './hooks.js';

const IMAGE_EXTENSIONS = ["jpg", "jpeg", "png", "webp"];
const VIDEO_EXTENSIONS = ["mp4", "webm", "ogg"];

const CurrentCard: React.FC<{ card: ParsedCard }> = ({ card }) => {

  const app = useApp()

  const createMediaElement = (key: string, url: string, extension: string) => {
    if (IMAGE_EXTENSIONS.includes(extension)) {
      return <img key={key} src={url} />;
    } else if (VIDEO_EXTENSIONS.includes(extension)) {
      return <video key={key} src={url} controls />;
    }
    return null;
  }

  const createTextElement = (key: string, text: string) => {
    return <span key={key}>{text}</span>;
  }

  const processAnnotation = (annotation: Annotation, i: number, text: string, lastEndPos: number) => {
    const cardElements: JSX.Element[] = [];
    cardElements.push(createTextElement(`t-{i}`, text.slice(lastEndPos, annotation.startPos)));

    const urlParts = String(annotation.url || "").split("|");
    const url = urlParts[0];
    const size = urlParts.length > 1 ? urlParts[1] : null;

    const extension = url.split('.').pop()?.toLowerCase() || IMAGE_EXTENSIONS[0];

    if (url.startsWith('http')) {
      const mediaElement = createMediaElement(`m-{i}`, url, extension);
      if (mediaElement) {
        cardElements.push(mediaElement);
      }
    } else if (app) {
      const mediaPath = app.metadataCache.getFirstLinkpathDest(url, ".")?.path;
      if (mediaPath) {
        const mediaResource = app.vault.adapter.getResourcePath(mediaPath);
        const mediaElement = createMediaElement(`m-{i}`, mediaResource, extension);
        if (mediaElement) {
          cardElements.push(mediaElement);
        }
      }
    }
    return { cardElements, endPos: annotation.endPos || 0 };
  }

  const textWithImages = React.useMemo(() => {
    let text = card.text;
    let annotations = card.annotations.filter((annotation): annotation is Annotation => annotation.type === 'image');

    annotations.sort((a, b) => (a.startPos || 0) - (b.startPos || 0));

    let cardElements: JSX.Element[] = [];
    let lastEndPos = 0;
    for (let i = 0; i < annotations.length; i++) {
      const annotation = annotations[i];
      const processed = processAnnotation(annotation, i, text, lastEndPos);
      cardElements = [...cardElements, ...processed.cardElements];
      lastEndPos = processed.endPos;
    }
    cardElements.push(createTextElement('t-last', text.slice(lastEndPos)));

    return cardElements;
  }, [card]);

  console.log('annotations!', card.annotations);
  return (
    <div className="card">
      <h2 className="card-text">{textWithImages}</h2>
    </div>
  );
}

export default CurrentCard;