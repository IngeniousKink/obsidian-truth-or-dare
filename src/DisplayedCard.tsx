import * as React from 'react'
import type { Card } from './parse-template.js';

// <div class="bg-white">
//   <div class="flex justify-center items-center min-h-screen min-w-full">
//         <div class="flex relative">
//             <div class="w-72 h-40 bg-green-400 transform transition-all skew-x-12 -skew-y-12 absolute rounded-lg">

//             </div>
//             <div class="w-72 h-40 bg-yellow-400 transform transition-all skew-x-12 -skew-y-12 absolute -top-4 -left-4 rounded-lg">
    
//             </div>
//             <div class="w-72 h-40 bg-red-400 transform transition-all skew-x-12 -skew-y-12 absolute -top-8 -left-8 rounded-lg">
    
//             </div>
//             <div class="w-72 h-40 bg-black transform transition-all skew-x-12 -skew-y-12 absolute -top-12 -left-12 rounded-lg">
    
//             </div>
//             <div class="w-72 h-40 bg-purple-400 transform transition-all skew-x-12 -skew-y-12 absolute -top-16 -left-16 rounded-lg">
//             </div>
//             <div class="w-72 h-40 bg-white flex justify-center items-center border-2 border-black transform transition-all skew-x-12 -skew-y-12 absolute -top-20 -left-20 rounded-lg">
//               Hello there! This is your crad<br>
//               foo
//                             Hello there! This is your crad<br>
//               foo
//                             Hello there! This is your crad<br>
//               foo

//               foo
//                             Hello there! This is your crad<br>
//               foo
//                             Hello there! This is your crad<br>
//               foo
//             </div>
//         </div>
//     </div>
// </div>

export const CurrentCard: React.FC<{ card: Card; }> = ({ card }) => (
  <div className="card">
    <h2 className="card-text">{card.text}</h2>
    {card.category && <h3 className="card-category">{card.category}</h3>}
  </div>
);

export const NoCard: React.FC = () => (
  <p>There is no card to display (yet).</p>
);

export const DisplayedCard: React.FC<{ card: Card | null; }> = ({ card }) => (
  card ? <CurrentCard card={card}/> : <NoCard/>
);