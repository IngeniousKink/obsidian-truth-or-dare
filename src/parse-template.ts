import type { Root, Html, Heading, List, ListItem, Paragraph, Text, PhrasingContent, Break } from "mdast";
import { parseCard } from "./parse-card.js";

// Define a Card type with a reference and text
export type Card = { ref: string; text: string; category?: string };

// Define a Stack type with a name, reference and an array of cards
export type Stack = { name: string; ref: string; cards: Card[]; };

// Define a GameTemplate type with an array of stacks
export type GameTemplate = { stacks: Stack[]; };
// ChildNode type can be either a Heading or a List
type ChildNode = Heading | List;

/**
 * This function takes a markdown abstract syntax tree (AST) and converts it into a game template.
 * The game template is represented as a GameTemplate, where each stack is an object with a name, reference and an array of cards.
 * Each heading in the markdown creates a new stack in the GameTemplate, and each list item under that heading
 * creates a new card in that stack.
 * 
 * For example, given the following markdown:
 * 
 * # Heading 1
 * - Item 1
 * - Item 2
 * 
 * # Heading 2
 * - Item 3
 * 
 * The resulting GameTemplate would be:
 * 
 * {
 *   stacks: [
 *     {
 *       name: "Heading 1",
 *       ref: "#Heading 1",
 *       cards: [
 *         { ref: "#Heading 1^0", text: "Item 1" },
 *         { ref: "#Heading 1^1", text: "Item 2" }
 *       ]
 *     },
 *     {
 *       name: "Heading 2",
 *       ref: "#Heading 2",
 *       cards: [
 *         { ref: "#Heading 2^0", text: "Item 3" }
 *       ]
 *     }
 *   ]
 * }
 */
export function convertMarkdownToGameTemplate(node: Root): GameTemplate {
  return node.children.reduce<GameTemplate>(
    (gameTemplate, child) => processMarkdownNode(gameTemplate, child as ChildNode),
    { stacks: [] }
  );
}

// Process a markdown node, which can either be a heading or a list
function processMarkdownNode(gameTemplate: GameTemplate, child: ChildNode): GameTemplate {
  if (child.type === 'heading') {
    const headingText = extractHeadingText(child);
    gameTemplate.stacks.push({ name: headingText, ref: "#" + headingText, cards: [] });
    return gameTemplate;
  }

  if (child.type === 'list') {
    let stack = getLastStack(gameTemplate);
    if (!stack) {
      stack = {
        name: "",
        ref: "",
        cards: extractCardsFromList(child, ""),
      };
      gameTemplate.stacks.push(stack);
    }
    stack.cards = extractCardsFromList(child, stack.ref);
    return gameTemplate;
  }

  return gameTemplate;
}

// Extract the text from a heading node
function extractHeadingText(node: Heading): string {
  const textNode = node.children[0] as Text;
  return textNode.value || '';
}

// Get the last stack from the game template
function getLastStack(gameTemplate: GameTemplate): Stack {
  return gameTemplate.stacks[gameTemplate.stacks.length - 1];
}

// Extract cards from a list node
function extractCardsFromList(listNode: List, stackRef: string): Card[] {
  let refCounter = 0;
  return listNode.children.reduce<Card[]>((cards, listItem: ListItem) => {
    const listItemCards = extractCardsFromListItem(listItem, refCounter++, stackRef);
    return [...cards, ...listItemCards];
  }, []);
}

// Extract cards from a list item
function extractCardsFromListItem(listItem: ListItem, refCounter: number, stackRef: string): Card[] {
  return listItem.children.reduce<Card[]>((cards, paragraph: Paragraph | Html) => {
    const paragraphCard = extractCardFromParagraph(paragraph, refCounter, stackRef);
    return [...cards, paragraphCard];
  }, []);
}

function extractCardFromParagraph(paragraph: Paragraph | Html, refCounter: number, stackRef: string): Card {
  let children = [];

  if (paragraph.type == 'html') {
    children = [paragraph];
  } else {
    children = paragraph.children;
  }

  let textValue = children.reduce((acc, textNode: Text | Html) => {
    if (textNode.type !== 'text' && textNode.type !== 'html') return acc;
    return acc + textNode.value;
  }, '');

  return {
    ref: stackRef + '^' + refCounter,
    text: textValue,
    ...{
      ...parseCard(textValue),
    }
  };
}