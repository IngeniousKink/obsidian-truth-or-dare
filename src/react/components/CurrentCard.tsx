import * as React from 'react';
import { ParsedCard, Annotation } from '../../parse/parse-card.js';
import { useMediaResource } from '@obsidian-truth-or-dare/hooks.js';

const IMAGE_EXTENSIONS = ["jpg", "jpeg", "png", "webp"];
const VIDEO_EXTENSIONS = ["mp4", "webm", "ogg"];

const createMediaElement = (key: string, url: string, extension: string) => {
  if (IMAGE_EXTENSIONS.includes(extension)) {
    return <img key={key} src={url} />;
  } else if (VIDEO_EXTENSIONS.includes(extension)) {
    return <video key={key} src={url} controls />;
  }
  return null;
}

const createTextElement = (key: string, text: string) => {
  return <span key={key}>{text}</span>;
}

const CurrentCard: React.FC<{ card: ParsedCard }> = ({ card }) => {
  const getMediaResource = useMediaResource();

  const processAnnotation = (annotation: Annotation, i: number, text: string, lastEndPos: number) => {
    const cardElements: JSX.Element[] = [];
    cardElements.push(createTextElement(`t-${i}`, text.slice(lastEndPos, annotation.startPos)));

    const urlParts = String(annotation.url || "").split("|");
    const url = urlParts[0];
    const extension = url.split('.').pop()?.toLowerCase() || IMAGE_EXTENSIONS[0];

    const mediaResource = getMediaResource(url);
    if (mediaResource) {
      const mediaElement = createMediaElement(`m-${i}`, mediaResource, extension);
      if (mediaElement) {
        cardElements.push(mediaElement);
      }
    }

    return { cardElements, endPos: annotation.endPos || 0 };
  }

  const textWithImages = React.useMemo(() => {
    let text = card.text;

    card.annotations.sort((a, b) => (a.startPos || 0) - (b.startPos || 0));

    let cardElements: JSX.Element[] = [];
    let lastEndPos = 0;
    for (let i = 0; i < card.annotations.length; i++) {
      const annotation = card.annotations[i];
      const processed = processAnnotation(annotation, i, text, lastEndPos);
      cardElements = [...cardElements, ...processed.cardElements];
      lastEndPos = processed.endPos;
    }
    cardElements.push(createTextElement('t-last', text.slice(lastEndPos)));

    return cardElements;
  }, [card]);

  return (
    <div className="card-stack">
      <div className="card"></div>
      <div className="card"></div>
      <div className="card"></div>
      <div className="card"></div>
      <div className="card"></div>
      <div className="card">
        <h2 className="card-text">{textWithImages}</h2>
      </div>
    </div>
  );
}

export default CurrentCard;
