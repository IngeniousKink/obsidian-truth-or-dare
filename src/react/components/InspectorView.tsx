import React from 'react';

import { JsonStringifyOutput } from './JsonStringifyOutput.js';
import { EventsDisplay } from './EventsDisplay.js';
import { RemainingCards } from './RemainingCards.js';
import { PreviousCards } from './PreviousCards.js';

import { 
  GameState, 
  getAvailableCards, 
  selectCardsByCategory, 
} from '../../gamestate.js';

import type { CardWithRef } from '../../parse/parse-template.js';

export interface InspectorViewProps {
  gameState: GameState,
}

export const InspectorView: React.FC<InspectorViewProps> = ({ gameState }: InspectorViewProps) => {
  const heading: string = "Truth or Dare";

  let cardsByCategory: { [x: string]: any; } = {};

  if (gameState.template) {
    cardsByCategory = selectCardsByCategory(gameState);
  }

  return (
    <div>
      <h1>Inspecting {heading}</h1>

      <h2>Status</h2>

      {gameState.previousCards && (
        <PreviousCards previousCards={gameState.previousCards} />
      )}
      {gameState.template && (<>
        <RemainingCards remainingCards={getAvailableCards(gameState)} />

        {gameState.events && (
          <>
            <h2>Events</h2>
            <EventsDisplay events={gameState.events} />
          </>
        )}
        {gameState.allocation && (
          <>
            <h2>Slots</h2>
            <JsonStringifyOutput javascriptObject={gameState.allocation} />
          </>
        )}
        {gameState.actors && (
          <>
            <h2>Players</h2>
            <JsonStringifyOutput javascriptObject={gameState.actors} />
          </>
        )}
        {gameState.template.stacks && (
          <>
            <h2>Template</h2>
            <JsonStringifyOutput javascriptObject={gameState.template.stacks} />
          </>
        )}

      </>)}
    </div>
  );
};
