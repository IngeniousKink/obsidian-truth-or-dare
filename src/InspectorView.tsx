import React from 'react';


import { DrawCardButton } from './DrawCardButton.js';
import { StacksDisplay } from './StacksDisplay.js';
import { EventsDisplay } from './EventsDisplay.js';
import { DisplayedCard } from './DisplayedCard.js';
import { RemainingCards } from './RemainingCards.js';
import { PreviousCards } from './PreviousCards.js';

import { 
  GameState, 
  getAvailableCards, 
  selectCardByRef, 
  selectCardsByCategory, 
  selectRandomAvailableCard, 
  selectRandomCard 
} from './gamestate.js';

import type { Card } from './parse-template.js';

export interface InspectorViewProps {
  gameState: GameState,
}

export const InspectorView: React.FC<InspectorViewProps> = ({ gameState }: InspectorViewProps) => {
  const heading: string = "Truth or Dare";

  let cardsByCategory: { [x: string]: any; } = {};

  if (gameState.template) {
    cardsByCategory = selectCardsByCategory(gameState);
  }

  return (
    <div>
      <h1>Inspecting {heading}</h1>

      <h2>Status</h2>

      {gameState.previousCards && (
        <PreviousCards previousCards={gameState.previousCards} />
      )}

      {gameState.template && (<>
        <RemainingCards remainingCards={getAvailableCards(gameState)} />

        {Object.keys(cardsByCategory).map(key => {
          const value: Card[] = cardsByCategory[key];
          const nextCard = selectRandomCard(value, gameState);
          return (
            <DrawCardButton
              key={key}
              categoryLabel={key}
              remainingCount={value.length}
              nextCard={nextCard} />
          );
        })}

        {gameState.events && (
          <>
            <h2>Events</h2>
            <EventsDisplay events={gameState.events} />
          </>
        )}
        {gameState.template.stacks && (
          <>
            <h2>Template</h2>
            <StacksDisplay stacks={gameState.template.stacks} />
          </>
        )}

      </>)}
    </div>
  );
};
