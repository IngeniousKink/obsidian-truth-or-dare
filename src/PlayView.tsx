import React from 'react';


import { DrawCardButton } from './DrawCardButton.js';
import { StacksDisplay } from './StacksDisplay.js';
import { EventsDisplay } from './EventsDisplay.js';
import { DisplayedCard } from './DisplayedCard.js';
import { RemainingCards } from './RemainingCards.js';
import { PreviousCards } from './PreviousCards.js';

import { 
  GameState, 
  getAvailableCards, 
  selectCardByRef, 
  selectCardsByCategory, 
  selectRandomAvailableCard, 
  selectRandomCard 
} from './gamestate.js';

import type { CardWithRef } from './parse-template.js';

export interface PlayViewProps {
  gameState: GameState,
}

export const PlayView: React.FC<PlayViewProps> = ({ gameState }: PlayViewProps) => {
  const heading: string = "Truth or Dare";

  let cardsByCategory: { [x: string]: any; } = {};

  if (gameState.template) {
    cardsByCategory = selectCardsByCategory(gameState);
  }

  return (
    <div>
      <h1>{heading}</h1>

      <DisplayedCard card={selectCardByRef(gameState, gameState.displayedCard)} />

      {gameState.template && (<>

        <h2>Buttons</h2>

        <DrawCardButton
          categoryLabel='random'
          remainingCount={getAvailableCards(gameState).length}
          nextCard={selectRandomAvailableCard(gameState)} />

        {Object.keys(cardsByCategory).map(key => {
          const value: CardWithRef[] = cardsByCategory[key];
          const nextCard = selectRandomCard(value, gameState);
          return (
            <DrawCardButton
              key={key}
              categoryLabel={key}
              remainingCount={value.length}
              nextCard={nextCard} />
          );
        })}

      </>)}
    </div>
  );
};
