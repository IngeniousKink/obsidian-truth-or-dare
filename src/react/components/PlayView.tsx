import React from 'react';
import { DrawCardButton } from './DrawCardButton.js';
import { DisplayedCard } from './DisplayedCard.js';

import {
  GameState,
  getAvailableCards,
  selectActorName,
  selectCardByRef,
  selectCardsByCategory,
  selectRandomAvailableCard,
  selectRandomCard
} from '../../gamestate.js';

import type { CardWithRef } from '../../parse/parse-template.js';
import { ActorList } from './ActorList.js';
import { CompleteCardButton } from './CompleteCardButton.js';

export interface GameStateWithDisplayedCard extends GameState {
  displayedCard: string;
}

interface PlayViewProps {
  gameState: GameStateWithDisplayedCard;
}

const PerformCard: React.FC<PlayViewProps> = ({ gameState }: PlayViewProps) => {
  const card = selectCardByRef(gameState, gameState.displayedCard);

  return (
    <>
      <DisplayedCard card={card} />
      <CompleteCardButton cardRef={gameState.displayedCard} />
    </>
  );
};


export const PlayView: React.FC<PlayViewProps> = ({ gameState }: PlayViewProps) => {
  let heading: string = "Truth or Dare";

  if (gameState.allocation?.length > 0 && gameState.allocation[0]) {
    heading += `, ${selectActorName(gameState, gameState.allocation[0])?.name}`;
  }

  heading += '?'

  let cardsByCategory: { [x: string]: any; } = {};

  if (gameState.template) {
    cardsByCategory = selectCardsByCategory(gameState);
  }

  const ChooseCategoryButtons = () => (
    <>
      <h2>Buttons</h2>
      <DrawCardButton
        categoryLabel='random'
        remainingCount={getAvailableCards(gameState).length}
        nextCard={selectRandomAvailableCard(gameState)} />
      {Object.keys(cardsByCategory).map(key => {
        const value: CardWithRef[] = cardsByCategory[key];
        const nextCard = selectRandomCard(value, gameState);
        return (
          <DrawCardButton
            key={key}
            categoryLabel={key}
            remainingCount={value.length}
            nextCard={nextCard} />
        );
      })}
    </>
  );

  const OutOfCards: React.FC = () => {
    return (
      <div>
        {'All suitable cards in this game have been played. Want to play another game ?'}
      </div>
    );
  };


  return (
    <div className="obsidian-truth-or-dare-container">
      <h1>{heading}</h1>


      {gameState.displayedCard
        ? <PerformCard gameState={gameState} />
        : (
          <div>
            {getAvailableCards(gameState).length > 0
              ? <ChooseCategoryButtons />
              : <OutOfCards />
            }
          </div>
        )
      }

      <br />
      <br />

      {gameState.template && <ActorList actors={gameState.actors} />}
    </div>
  );
};
