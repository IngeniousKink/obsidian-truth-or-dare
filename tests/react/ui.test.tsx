// @vitest-environment jsdom

import * as React from 'react'
import * as renderer from 'react-test-renderer';

import { CardWithRef } from '@obsidian-truth-or-dare/parse/parse-template.js';

import { NoCard, DisplayedCard } from '@obsidian-truth-or-dare/react/components/DisplayedCard.js'
import { GameStateWithDisplayedCard, PlayView } from '@obsidian-truth-or-dare/react/components/PlayView.js';
import { InspectorView } from '@obsidian-truth-or-dare/react/components/InspectorView.js';
import CurrentCard from '@obsidian-truth-or-dare/react/components/CurrentCard.js';
import { expect, test } from 'vitest';

const testCard : CardWithRef = {
   text: 'Do something.',
   ref: 'ref1',
   annotations: [],
};

test('renders NoCard', () => {
  const tree = renderer
    .create(<NoCard />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});


test('renders CurrentCard', () => {
  const tree = renderer
    .create(<CurrentCard card={testCard} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders DisplayedCard = card', () => {
  const tree = renderer
    .create(<DisplayedCard card={testCard} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders DisplayedCard = null', () => {
  const tree = renderer
    .create(<DisplayedCard card={null} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

const testGameState : GameStateWithDisplayedCard = {
  "actors": [],
  "allocation": [],
  "preview": {
    "allocation": [],
  },
  "template": {
      "stacks": [
          {
              "name": "first stack",
              "ref": "#first stack",
              "cards": [
                  {
                      "ref": "#first stack^0",
                      "text": "one card",
                      "annotations": [],
                  }
              ]
          },
          {
              "name": "second stack",
              "ref": "#second stack",
              "cards": [
                  {
                      "ref": "#second stack^0",
                      "text": "first of two cards",
                      "annotations": [],
                  },
                  {
                      "ref": "#second stack^1",
                      "text": "second of two cards",
                      "annotations": [],
                  }
              ]
          }
      ]
  },
  "events": [
      {
          "type": "draw_card",
          "timestamp": 1706983496470,
          "cardRef": "#first stack^0"
      }
  ],
  "previousCards": [],
  "seed": "{\"stacks\":[{\"name\":\"first stack\",\"ref\":\"#first stack\",\"cards\":[{\"ref\":\"#first stack^0\",\"text\":\"one card\"}]},{\"name\":\"second stack\",\"ref\":\"#second stack\",\"cards\":[{\"ref\":\"#second stack^0\",\"text\":\"first of two cards\"},{\"ref\":\"#second stack^1\",\"text\":\"second of two cards\"}]}]}[{\"type\":\"card-draw\",\"timestamp\":1706983496470,\"card\":\"#first stack^0\"}]",
  "displayedCard": "#first stack^0"
};

test('renders PlayView', () => {

  const tree = renderer
    .create(<PlayView gameState={testGameState} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders InspectorView', () => {
  const tree = renderer
    .create(<InspectorView gameState={testGameState} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});