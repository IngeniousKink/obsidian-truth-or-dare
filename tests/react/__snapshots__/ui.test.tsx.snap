// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders CurrentCard 1`] = `
<div
  className="card"
>
  <h2
    className="card-text"
  >
    Do something.
  </h2>
</div>
`;

exports[`renders DisplayedCard = card 1`] = `
<div
  className="card"
>
  <h2
    className="card-text"
  >
    Do something.
  </h2>
</div>
`;

exports[`renders DisplayedCard = null 1`] = `
<p>
  There is no card to display (yet).
</p>
`;

exports[`renders DisplayedCard = null 2`] = `
<div>
  <h1>
    Truth or Dare
  </h1>
  <h2>
    Card
  </h2>
  <div
    className="card"
  >
    <h2
      className="card-text"
    >
      one card
    </h2>
  </div>
  <h2>
    Status
  </h2>
  <p>
    So far, 
    0
     cards have been played.
  </p>
  <p>
    There are 
    2
     cards remaining.
  </p>
  <h2>
    Buttons
  </h2>
  <h2>
    Events
  </h2>
  <p>
    In this game there are 
    1
     events played so far.
  </p>
  <h2>
    Template
  </h2>
  <pre
    style={
      {
        "fontSize": "80%",
      }
    }
  >
    [
  {
    "name": "first stack",
    "ref": "#first stack",
    "cards": [
      {
        "ref": "#first stack^0",
        "text": "one card"
      }
    ]
  },
  {
    "name": "second stack",
    "ref": "#second stack",
    "cards": [
      {
        "ref": "#second stack^0",
        "text": "first of two cards"
      },
      {
        "ref": "#second stack^1",
        "text": "second of two cards"
      }
    ]
  }
]
  </pre>
  ;
</div>
`;

exports[`renders DisplayedCard 1`] = `
<div
  className="card"
>
  <h2
    className="card-text"
  >
    Do something.
  </h2>
</div>
`;

exports[`renders DisplayedCard 2`] = `
<p>
  There is no card to display (yet).
</p>
`;

exports[`renders NoCard 1`] = `
<p>
  There is no card to display (yet).
</p>
`;

exports[`renders correctly 1`] = `
<p>
  There is no card to display (yet).
</p>
`;
